sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java

```diff
@@ -717,4 +740,123 @@ private void throwOnAnonymousAccess() {
                 + "credentials"));
         }
     }
+
+    /**
+     * Restores a previously deleted container.
+     * If the container associated with provided <code>deletedContainerName</code>
+     * already exists, this call will result in a 409 (conflict).
+     * This API is only functional if Container Soft Delete is enabled
+     * for the storage account associated with the container.
+     * For more information, see the
+     * <a href="TBD">Azure Docs</a>. TODO (kasobol-msft) add link to REST API docs
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.blob.BlobServiceAsyncClient.undeleteBlobContainer#String-String}
+     *
+     * @param deletedContainerName The name of the previously deleted container.
+     * @param deletedContainerVersion The version of the previously deleted container.
```

Comment:
Unless the rest docs already say this, maybe we should add where this version can be retrieved from

Reply:
We don't have rest docs yet so I'd hold on with that. However I've showcased how to get version id in the sample snippets. I think that should be enough.